src/GT.UI/Controllers/ListingsController.cs

```diff
@@ -10,50 +12,59 @@ namespace GT.UI.Controllers
 	[ApiController]
 	public class ListingsController : ControllerBase
 	{
-		private readonly GTAppContext _context;
+		private readonly IGTListingService _ListingService;
+		private readonly UserManager<IdentityUser> _userManager;
 
-		public ListingsController(GTAppContext context)
-		{
-			_context = context;
+
+		public ListingsController(IGTListingService iGTListingService, UserManager<IdentityUser> userManager)
+{
+			_ListingService = iGTListingService;
+			_userManager = userManager;
 		}
 
 		// GET: api/Listings
 		[HttpGet]
 		public async Task<ActionResult<IEnumerable<Listing>>> GetListings()
 		{
-			return await _context.Listings.ToListAsync();
+			var listings = await _ListingService.GetAsync();
+
+			if(listings == null)
+			{
+				return NotFound();
+			}
+
+			return Ok(listings);
 		}
 
 		// GET: api/Listings/5
 		[HttpGet("{id}")]
 		public async Task<ActionResult<Listing>> GetListing(string id)
 		{
-			var listing = await _context.Listings.FindAsync(id);
+			var listing = await _ListingService.GetByIdAsync(id);
 
 			if (listing == null)
 			{
 				return NotFound();
 			}
 
-			return listing;
+			return Ok(listing);
 		}
 
 		// PUT: api/Listings/5
 		// To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
 		[HttpPut("{id}")]
-		public async Task<IActionResult> PutListing(string id, Listing listing)
+		public async Task<IActionResult> PutListing(string id, ListingDTO listing)
 		{
 			if (id != listing.Id)
 			{
 				return BadRequest();
 			}
 
-			_context.Entry(listing).State = EntityState.Modified;
-
 			try
 			{
-				await _context.SaveChangesAsync();
+				await _ListingService.UpdateAsync(listing, id);
```

Comment:
listingService should return correct object - should not await new DB get by Id (null in case of Error handling)

Reply:
Could please explain a bit more. Do you expect the UpdateAsync to return the updated DTO?